modules/shared/src/main/scala/org/tessellation/schema/currencyMessage.scala

```diff
@@ -60,4 +62,21 @@ object currencyMessage {
   case class CurrencyMessage(messageType: MessageType, address: Address, parentOrdinal: MessageOrdinal) {
     def ordinal: MessageOrdinal = parentOrdinal.next
   }
+
+  def fetchStakingAddress(state: CurrencySnapshotInfo): Option[Address] =
+    state.lastMessages
+      .flatMap(_.view.filterKeys(_ == MessageType.Staking).values.maxByOption(_.ordinal))
+      .map(_.address)
```

Comment:
Why is it implemented as above instead of this?
```
state.lastMessages
  .flatMap(_.get(MessageType.Staking))
  .map(_.address)
```
Is there a plan to change the implementation so that the values are a list?

Reply:
I guess my mind was fixed on the idea that there may be multiple messages of the same type in the snapshot (but with increasing ordinal) and hence I reflected that here as well - which (you are right) doesn't make sense for the Map having `MessageType` as a key. Will change the code to your implementation. Thanks

Reply:
Thank you @TheMMaciek . I'm doing something similar in my PR so I'll copy your code there.